tutorials/yes-no-game/full game/yesnogame.edn

```diff
@@ -0,0 +1,251 @@
+  0 >= .right-images-count
+
+  .cat-images >= .left-images
+  .cat-images >= .right-images
+  true >= .same-images
+  true >= .new-round
+  0 >= .new-round-number
+
+  ; variables to reset each round
+  1 >= .current-round
+  0 >= .seconds-elapsed
+  0 >= .total-score
+  false >= .yes-input
+  false >= .input-received
+  false >= .game-over)
+
+(defshards reset-values []
+
+  false > .yes-input
+  false > .input-received
+  false > .game-over
+  1 > .current-round
+  0 >  .seconds-elapsed
+  0 > .total-score)
+
+; determines the images used for the round
+(defshards initialize-round []
+
+  0 > .seconds-elapsed
+  
+  ; determine if the images on the left and right will be the same or different
+  (If
+   :Predicate (-> (RandomInt :Max 2) (Is 0))
+   :Then (-> true > .same-images)
+   :Else (-> false > .same-images))
+
+  ; gets the image series to be used for the left side
+  (RandomInt :Max .total-image-series) > .left-series-index
+
+  ; get the number of images in the series to be used for the left
+  .all-images (Take .left-series-index) > .left-images
+  (Count .left-images) > .left-images-count
+
+  ; setup for the right image
+  (If
+   :Predicate (-> .same-images (Is true))
+   ; if images should be from the same series
+   :Then (->
+          .left-series-index > .right-series-index
+          (RandomInt :Max .left-images-count) > .left-image-index
+          (RandomInt :Max .left-images-count) > .right-image-index
+          (Repeat
+           :Action (-> (RandomInt :Max .left-images-count) > .right-image-index)
+           :Forever true
+           :Until (-> .right-image-index (IsNot .left-image-index)))
+
+          .right-image-index (IsNot .left-image-index) (Log "Are they not the same: "))
+
+   ; if images should be from a different series   
+   :Else (->
+          (RandomInt :Max .total-image-series) > .right-series-index
+          (Repeat
+           :Action (-> (RandomInt :Max .total-image-series) > .right-series-index)
+           :Forever true
+           :Until (-> .right-series-index (IsNot .left-series-index)))
+
+          ; get the number of images in the series to be used for the right 
+          .all-images (Take .right-series-index) > .right-images
+          (Count .right-images) > .right-images-count
+          (RandomInt :Max .left-images-count) > .left-image-index
+          (RandomInt :Max .right-images-count) > .right-image-index))
+
+  false > .new-round
+  false > .input-received)
+
+; creates an UI.Area with the left image
+(defshards left-image [image]
+  (UI.Area
+   :Position (float2 left-x-offset, 0.0)
+   :Anchor Anchor.Center
+   :Contents
+   (-> image (UI.Image :Scale (float2 0.5)))))
+
+; creates an UI.Area with the right image
+(defshards right-image [image]
+  (UI.Area
+   :Position (float2 right-x-offset, 0.0)
+   :Anchor Anchor.Center
+   :Contents
+   (-> image (UI.Image :Scale (float2 0.5)))))
+
+; creates the main game area
+(defshards main-game-ui []
+  (UI.TopPanel
+   (->
```

Comment:
If you build from the latest `devel` branch, this won't work as panels now have additional parameters and `:Contents` isn't the first one.

Consider changing it to:
```suggestion
  (UI.TopPanel
   :Contents
   (->
```

Note: this applies to other panels in the same file, except `UI.CentralPanel` but let's have it there as well for consistency.